R/shared-methods.R

```diff
@@ -509,6 +509,158 @@ DataClass <- R6::R6Class(
         source_data_cols = paste(unlist(self$source_data_cols), collapse = ", ")
       )
       return(sum_df)
+    },
+
+    #' @description Run tests on data class. Inherited by child classes so test
+    #' run through each class.
+    #' @param download logical. To download the data (TRUE) or use a snapshot
+    #' (FALSE). Defaults to FALSE.
+    #' @importFrom purrr walk map
+    #' @importFrom dplyr slice_tail
+    #' @importFrom testthat test_that expect_s3_class expect_true expect_error
+    test = function(download = FALSE) {
+      data_name <- paste0(class(self)[1], " at level ", self$level)
+      raw_path <- paste0(
+        "custom_data/", class(self)[1],
```

Comment:
I think we want this path to be user configurable and default to a temporary directory to pass CRAN.

Reply:
I've made this user changable but currently it defaults to the custom/data directory. What would I call a tmp directory for cran? I imagine we would do this by having an on CRAN condition which would pass different location?